dev-db/mongodb: bump to 3.6.1

hydrapolic opened the pull request.

ultrabug commented:
Thanks mate, I'm testing this right now.

But as I told you, I'll be tree cleaning old versions before merging this. I'd like to have only two majors supported on portage.

hydrapolic commented:
Thanks.

It would be nice to keep 3.2 for a little while :)

hydrapolic commented:
That said,I'm ok with dropping 2.6/3.0.

ultrabug commented:
Please make sure that the `install_compass` is NOT installed because it's misleading and not supporting gentoo.

Also, since I have the `splitdebug` FEATURE on, it leads to a 14G tmp directory! I wonder if a RESTRICT="splitdebug" would prevent this for this package... but that's my user's choice so I guess it's okay.

hydrapolic commented:
Ok I'll check again on compass, thanks for noticing. About the splitdebug, should we warn the users that it takes some space? Well I can imagine that in some cases you'll run this in production if you debug some obscure crashes that are hard to reproduce in testing environment.

hydrapolic commented:
@ultrabug, removed compass, bumped required versions (gcc,scons), added python deps (pyyaml,typing), also removed bundled libs. Tests pass on amd64, please give it a try ;)

I've tried to unbundle wiredtiger, but no luck so far. Also using system-icu causes the tests to fail.

SoapGentoo commented on app-admin/mongo-tools/mongo-tools-3.6.0.ebuild:
```diff
@@ -0,0 +1,63 @@
+	# ensure we use bash wrt #582906
+	sed -i 's@/bin/sh@/bin/bash@g' build.sh || die
+
+	default
```
In general, prefer to call `default` first thing in `src_prepare`

SoapGentoo commented on app-admin/mongo-tools/mongo-tools-3.6.0.ebuild:
```diff
@@ -0,0 +1,63 @@
+
+EAPI=6
+
+inherit eutils
```
do you still need this?

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+		libressl? ( dev-libs/libressl:0= )
+	)"
+DEPEND="${RDEPEND}
+	dev-python/cheetah
```
`[${PYTHON_USEDEP}]`

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+	)"
+DEPEND="${RDEPEND}
+	dev-python/cheetah
+	dev-python/pyyaml
```
`[${PYTHON_USEDEP}]`

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+DEPEND="${RDEPEND}
+	dev-python/cheetah
+	dev-python/pyyaml
+	dev-python/typing
```
`[${PYTHON_USEDEP}]`

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+	dev-python/cheetah
+	dev-python/pyyaml
+	dev-python/typing
+	=dev-lang/python-2*
```
`${PYTHON_DEPS}` always, never depend on naked python impls

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+	dev-python/typing
+	=dev-lang/python-2*
+	<dev-util/scons-3
+	>=sys-devel/gcc-5.4.0:*
```
pointless dependency, remove

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+	debug? ( dev-util/valgrind )
+	kerberos? ( dev-libs/cyrus-sasl[kerberos] )
+	test? (
+		dev-python/pymongo
```
`[${PYTHON_USEDEP}]`

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+			--cpppath="${EPREFIX}/usr/include"
+			--libpath="${EPREFIX}/usr/$(get_libdir)"
+		)
+	fi
```
I currently don't see why this is in a prefix conditional, maybe add it to the global option array

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+	# --use-system-tcmalloc is strongly NOT recommended:
+	# https://www.mongodb.org/about/contributors/tutorial/build-mongodb-from-source/
+
+	scons_opts=(
```
do all the `scons_opts` mucking `src_configure`

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+
+	if use debug; then
+		scons_opts+=( --dbg=on )
+	fi
```
use the more concise syntax: `use debug && scons_opts+=( --dbg=on )`

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+	if ! use debug; then
+		filter-flags '-m*'
+		filter-flags '-O?'
+	fi
```
do this in `src_configure` after the `scons_opts` mucking

SoapGentoo commented on dev-db/mongodb/mongodb-3.6.0.ebuild:
```diff
@@ -0,0 +1,184 @@
+# FEATURES="test -usersandbox" emerge dev-db/mongodb
+src_test() {
+	"${EPYTHON}" ./buildscripts/resmoke.py --dbpathPrefix=test --suites core --jobs=$(makeopts_jobs) || die "Tests failed"
+}
```
`src_test` always between `src_compile` and `src_install`

ultrabug commented:
Thanks @hydrapolic  mate, please address the review of @SoapGentoo and directly bump to 3.6.1 so I can test as well afterwards

hydrapolic commented:
Yes I will,but I'm taking a break until the very first days of 2018 :) thanks @SoapGentoo for all the reviews!

ultrabug commented:
np @hydrapolic, happy new year ;)

Meanwhile, I'll start cleaning 2.6 and 3.0 versions from tree

hydrapolic commented:
@ultrabug, pushed up version 3.6.1, test pass on amd64, please give it a try. Thank you!

hydrapolic commented:
And happy new year to you too!

ultrabug commented:
Thank you mate, testing now.

ultrabug commented:
Very good mate, works great for me! Thanks a lot 👍

hydrapolic commented:
Thank you for the help guys!

cjwijtmans commented:
any chance yet of getting mongodb/scons to python3? The only packagaes on my server on python2 atm.

hydrapolic commented:
This is currently a mongodb limitation,please ask for python3 support upstream. We will be more than happy to add py3 support in gentoo then :)
